README.md

```diff
@@ -34,4 +35,91 @@ qmake
+* **holosqape** contains the QML based GUI project wich compiles to an executable that can be build for all platforms supported by Qt. It manages installed Holochain apps and currently sports a generic UI that creates widgets and buttons for each zome function an app implements automatically. It also includes a JavaScript based console in which apps can be instantiated.
+* **holoconsole** this builds to another executable that basically hosts the same functionality as holosqape but without GUI. Instead it has some compatibility shims to enable running JavaScript libraries such as [Tape](https://github.com/substack/tape). It's main purpose is hosting an environment for app tests, including scenario integration tests.
+* **app-test-scaffold** this is an example showing how to build an app test script to run with holoconsole.
+
+
+
+## App tests
+
+### Binding objects
+In order to test a Holochain app we need a JS script that we run in holoconsole. The script engine adds a global object ```Container``` to the context before loading and running the user script. It currently has the following methods:
+
+* installedApps()
+* instantiate(dnaHash)
+* installApp(filename)
+
+and more as can be seen in [container.h](bindings/container.h).
+
+instantiate() returns an ```App``` object as defined in [app.h](bindings/app.h) with a method
+
+* call(zome_name, capability_name, function_name, parameters)
+
+(amongst others) which calls the specified zome function, blocks, and returns the result.
+
+This enables us to write code like this:
+
+```javascript
+let dnaHash = Container.installedApps()[0]
+let app = Container.instantiate(dnaHash)
+// app is actually a C++ object that wraps the Rust Holochain instance
+app.name() // -> App name defined in DNA
+app.start() // starts the instance's event loop and later will start the network node and DHT (not implement in Rust yet)
+let result = app.call("zome name", "capability name", "function name", "parameters") // runs the zome function which invovles interpreting the app's WASM code
+```
+
+
+### Using Tape testing framework
+To get proper assertions and formatted output we want to use existing JS scripting frameworks. We chose Tape for now because of its mininmal footprint.
+
+In order to build a JS bundle including Tape, we use Webpack. ```app-test-scaffold``` contains a working example which can be run like this:
+
+```
+cd app-test-scaffold
+npm install
+webpack-cli
+```
+You should now find the output in ```app-test-scaffold/dist/bundle.js```.
+Compile holoconsole and run it like this:
+```
+./holoconsole <path-to-repo>/app-test-scaffold/dist/bundle.js
+```
+
+It should give you the following output:
+
+```
+$ ./holoconsole ~/code/HoloSqape/app-test-scaffold/dist/bundle.js
```

Comment:
caps problem?

Reply:
In what way?

Reply:
If @luckus is on osx and @zippy is on Linux, well, Linux is case sensitive and osx is not.  Since CI is Linux, is CI having "case" issues like Zippy?